x-pack/heartbeat/scenarios/framework/fakeloader.go

```diff
@@ -27,29 +26,33 @@ func newLoaderDB() *loaderDB {
 	}
 }
 
-func (ldb loaderDB) AddState(sf stdfields.StdMonitorFields, state *monitorstate.State) {
+func (ldb *loaderDB) AddState(sf stdfields.StdMonitorFields, state *monitorstate.State) {
 	ldb.mtx.Lock()
 	defer ldb.mtx.Unlock()
 
-	ldb.lastTime = time.Now()
-
-	ldb.keysToState[monitorstate.LoaderDBKey(sf, ldb.lastTime, 0)] = state
+	key := keyFor(sf)
+	ldb.keysToState[key] = state
 }
 
-func (ldb loaderDB) GetState(sf stdfields.StdMonitorFields) *monitorstate.State {
+func (ldb *loaderDB) GetState(sf stdfields.StdMonitorFields) *monitorstate.State {
 	ldb.mtx.Lock()
 	defer ldb.mtx.Unlock()
```

Comment:
I know this change isn't part of this PR, but you could make `loaderDb.mtx` a `sync.RWMutex` and use `ldb.mtx.RLock()` and `ldb.mtx.RUnlock()` here.

Reply:
It's a good point, but this isn't performance critical code since it's just for testing.